File: 02_question/question2.py
class Mouth:
    "Mouth"

    def __init__(self, person, mouth =  False):
        self.mouth = mouth
        self.person = person
  

    def open(self):
        "Open Mouth"

        self.mouth = True
        if self.mouth:
            print(self.person, ": 'o' AAAHHHHH")
            self.mouth = False
            print(self.person,": '-'")

      
  
    def close(self):
        self.mouth = False
        print(self.person, ":Who are you my doctor?  '-'")
  
    def askToOpen(self, person):
        print(person.occupation,  ':  Open your mouth please.')
        if person.occupation == 'doctor':
            print(self.person, "Okay Doc!")
            self.open()
        else:
            self.mouth = False
            print(self.person, ":Who are you my doctor?  '-' ")
   

m = Mouth('David')
  

  



class Person:
    "Person Occupation"
    def __init__(self, occupation):
      self.occupation = occupation


d = Person('doctor')
t = Person('teacher')
l = Person('lawyer')


m.askToOpen(l)
print('=========')
m.askToOpen(t)
print('=========')
m.askToOpen(d)

File: 02_question/question2.js
class Mouth{
  constructor(person, mouth = false){
    this.mouth = mouth
    this.person = person
  }

  open(){
    this.mouth = true
      if (this.mouth){
        console.log(`${this.person}: 'O' AHHHHH`);

        setTimeout(() => {
          this.mouth = false
          console.log(`${this.person}: '-'`);
        }, 3000)
      }
  }
  close(){
    this.mouth = false;
    console.log(`${this.person}: Who are you my doctor?  '-' `);
  }
  askToOpen(person){
    console.log(`${person.occupation}: Open your mouth please.`);
    if(person.occupation === 'doctor'){
      console.log(`${this.person}: Okay Doc!`);
      this.open()
    } else {
      this.close()
    }
   
    
  }

  

}

class Person{
  constructor(occupation){
    this.occupation = occupation.toLowerCase();
  }
}

let d = new Person('Doctor');
let t = new Person('Teacher');
let l = new Person('lawyer');
let m = new Mouth('David');

m.askToOpen(t)

m.askToOpen(d);